example/sparse/linear_classification.py

```diff
@@ -56,19 +81,34 @@
     kvstore = args.kvstore
     batch_size = args.batch_size
     optimizer = args.optimizer
-
+    log_interval = args.log_interval
+    lr = args.lr
     # create kvstore
     kv = mx.kvstore.create(kvstore) if kvstore else None
     rank = kv.rank if kv else 0
     num_worker = kv.num_workers if kv else 1
 
     # dataset
-    num_features = AVAZU['num_features']
-    data_dir = os.path.join(os.getcwd(), 'data')
-    train_data = os.path.join(data_dir, AVAZU['train'])
-    val_data = os.path.join(data_dir, AVAZU['test'])
-    get_libsvm_data(data_dir, AVAZU['train'], AVAZU['url'])
-    get_libsvm_data(data_dir, AVAZU['test'], AVAZU['url'])
+    if args.model == 'linear':
```

Comment:
Why is dataset dependent on the type of model?

Reply:
Other implementations (Keras, Tensorflow, etc.) use adult dataset and I think it's better to be comparable here. And feature engineering is important. Avazu libsvm dataset cannot be directly used in wide and deep model because of some feature engineering issues, for example, there is one format issue, features in one side (wide or deep) should be together.